clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractPartitionAssignor.java

```diff
@@ -56,13 +58,14 @@ public Subscription subscription(Set<String> topics) {
         for (Map.Entry<String, Subscription> subscriptionEntry : subscriptions.entrySet())
             allSubscribedTopics.addAll(subscriptionEntry.getValue().topics());
 
-        Map<String, Integer> partitionsPerTopic = new HashMap<>();
+        Map<String, List<PartitionInfo>> partitionsPerTopic = new HashMap<>();
         for (String topic : allSubscribedTopics) {
-            Integer numPartitions = metadata.partitionCountForTopic(topic);
-            if (numPartitions != null && numPartitions > 0)
-                partitionsPerTopic.put(topic, numPartitions);
-            else
+            List<PartitionInfo> partitionsForThisTopic = new ArrayList<>(metadata.partitionsForTopic(topic));
```

Comment:
Should we add null check on `metadata.partitionsForTopic(topic))`?

I tried this patch on my 0.10.2 branch and bunch of ConsumerCoordinatorTest failed due to NPE. The tests pass after the NPE is handled.